api/nodes/views.py

```diff
@@ -106,8 +118,132 @@ def get_queryset(self):
 
 
-class NodePointersList(generics.ListAPIView, NodeMixin):
+class NodePointersList(generics.ListCreateAPIView, NodeMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+    )
+
     serializer_class = NodePointersSerializer
 
     def get_queryset(self):
         return self.get_node().nodes_pointer
+
+
+class NodePointerDetail(generics.RetrieveDestroyAPIView, NodeMixin):
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+    )
+
+    serializer_class = NodePointersSerializer
+
+    # overrides RetrieveAPIView
+    def get_object(self):
+        pointer_lookup_url_kwarg = 'pointer_id'
+        pointer = get_object_or_404(Pointer, self.kwargs[pointer_lookup_url_kwarg])
+        return pointer
+
+    # overrides DestroyAPIView
+    def perform_destroy(self, instance):
+        user = self.request.user
+        auth = Auth(user)
+        node = self.get_node()
+        pointer = self.get_object()
+        node.rm_pointer(pointer, auth)
+        node.save()
+
+
+class NodeFilesList(generics.ListAPIView, NodeMixin):
+    serializer_class = NodeFilesSerializer
+
+    permission_classes = (
+        drf_permissions.IsAuthenticatedOrReadOnly,
+    )
+
+    def get_valid_self_link_methods(self, user, root_folder=False):
+        valid_methods = {'file': [], 'folder': [], }
+        if user is None:
+            return valid_methods
+
+        permissions = self.get_node().get_permissions(user)
+        if 'read' in permissions:
+            valid_methods['file'].append('GET')
+        if 'write' in permissions:
+            valid_methods['file'].append('POST')
+            valid_methods['file'].append('DELETE')
+            valid_methods['folder'].append('POST')
+            if not root_folder:
+                valid_methods['folder'].append('DELETE')
+
+        return valid_methods
+
+    @staticmethod
+    def get_file_item(item, valid_file_methods, node_id, cookie, obj_args):
+        file_item = {
+            'valid_self_link_methods': valid_file_methods[item['kind']],
+            'provider': item['provider'],
+            'path': item['path'],
+            'name': item['name'],
+            'node_id': node_id,
+            'cookie': cookie,
+            'args': obj_args,
+            'waterbutler_type': 'file',
+            'item_type': item['kind'],
+        }
+        if file_item['item_type'] == 'folder':
+            file_item['metadata'] = {}
+        else:
+            file_item['metadata'] = {
+                'content_type': item['contentType'],
+                'modified': item['modified'],
+                'size': item['size'],
+                'extra': item['extra'],
+            }
+        return file_item
+
+    def get_queryset(self):
+        query_params = self.request.query_params
+
+        addons = self.get_node().get_addons()
+        user = self.request.user
+        cookie = user.get_or_create_cookie() if self.request.user else None
```

Comment:
This shouldn't be necessary if OSF API can talk to waterbutler using a token/ticket. @icereval, that's the plan, yes?